dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/utils/MiscUtils.kt

```diff
@@ -93,6 +96,28 @@ object JsonSerializer {
             else -> JsonPrimitive(item.toString())
         }
     }
+
+    /**
+     * This method will convert all values to JSON in a safe way, meaning if serialization fails
+     * for the particular value, the process will continue and faulty value will be dropped.
+     */
+    @InternalApi
+    fun Map<String, Any?>.safeMapValuesToJson(internalLogger: InternalLogger): Map<String, Any?> {
+        val result = mutableMapOf<String, Any?>()
```

Comment:
> [!NOTE]
> Maybe we could mark the return value as `Map<String, JsonElement>` instead.

Reply:
Indeed, good point. It should still be compatible with `Map<String, Any?>` type required by the `additionalProperties` normally.